spec/controllers/products_controller_spec.rb

```diff
@@ -109,21 +109,47 @@
 
   describe '#destroy' do
     product = FactoryBot.create(:product)
+    count_deleted_products = Product.where(status: 'deleted').count
 
-    it 'return a found response' do
-      delete :destroy, params: { id: product }
-      expect(response).to have_http_status(:found)
+    it 'return status deleted for some product' do
+      product_ids = []
+
+      5.times do
+        product_ids.push(FactoryBot.create(:product).id)
+      end
```

Comment:
you can use create_list instead of this iterator https://www.rubydoc.info/gems/factory_bot/FactoryBot%2FSyntax%2FMethods:create_list